src/niscope/system_tests/test_system_niscope.py

```diff
@@ -484,4 +339,228 @@ def test_configure_trigger_window(self, multi_instrument_session):
+        # Initialize with NaN so we can later verify all samples were overwritten by the driver.
+        waveform.fill(float('nan'))
+        multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
+        multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records_to_acquire, True)
+        with multi_instrument_session.initiate():
+            waveforms = multi_instrument_session.channels[test_channels].fetch_into(
+                waveform=waveform,
+                record_number=test_starting_record_number,
+                num_records=test_num_records_to_fetch)
+
+        for sample in waveform:
+            assert not math.isnan(sample)
+        assert len(waveforms) == test_num_channels * test_num_records_to_fetch
+
+        expected_channels = test_channels.split(',') * test_num_records_to_fetch
+        expected_records = [2, 2, 3, 3, 4, 4]
+        for i in range(len(waveforms)):
+            record_wfm = waveforms[i].samples
+            assert len(record_wfm) == test_record_length
+            for j in range(len(record_wfm)):
+                assert record_wfm[j] == waveform[i * test_record_length + j]
+            assert waveforms[i].channel == expected_channels[i]
+            assert waveforms[i].record == expected_records[i]
+
+    def test_error_message(self):
+        try:
+            # We pass in an invalid model name to force going to error_message
+            with niscope.Session('FakeDevice', False, True, 'Simulate=1, DriverSetup=Model:invalid_model; BoardType:PXIe'):
+                assert False
+        except niscope.Error as e:
+            assert e.code == -1074118609
+            assert e.description.find('Simulation does not support the selected model and board type.') != -1
+
+    def test_get_self_cal_last_date_time(self, single_instrument_session):
+        last_cal = single_instrument_session.get_self_cal_last_date_and_time()
+        assert last_cal.month == 12
+        assert last_cal.day == 21
+        assert last_cal.year == 1999
+        assert last_cal.hour == 0
+        assert last_cal.minute == 0
+
+    def test_get_ext_cal_last_date_time(self, single_instrument_session):
+        last_cal = single_instrument_session.get_ext_cal_last_date_and_time()
+        assert last_cal.month == 12
+        assert last_cal.day == 21
+        assert last_cal.year == 1999
+        assert last_cal.hour == 0
+        assert last_cal.minute == 0
+
+    def test_get_self_cal_last_temperature(self, single_instrument_session):
+        last_cal_temp = single_instrument_session.get_self_cal_last_temp()
+        assert last_cal_temp == 25
+
+    def test_get_ext_cal_last_temperature(self, single_instrument_session):
+        last_cal_temp = single_instrument_session.get_ext_cal_last_temp()
+        assert last_cal_temp == 25
```

Comment:
@danestull please add a TODO so that we don't forget to move these tests back after the grpc-device change goes in.

Reply:
added this TODO